redhat-appstudio/service-provider-integration-operator
# custom user oauth configuration

### What does this PR do?
makes it possible to configure oauth service using secret in same namespace as SPIAccessTokenBinding

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-205

### How to test this PR?


images:
```
quay.io/mvala/service-provider-integration-operator:svpi205-userOauth
quay.io/mvala/service-provider-integration-oauth:svpi205-userOauth
```

!!! make sure token you're using to do oauth flow has list and get secret perms in token's namespace

1. deploy with some nonsense oauth configuration for one of service providers
2. create spiaccesstokenbinding and try to do oauth flow and see it fail
3. create secret with oauth configuration in same namespace as a token (see `samples/sp-secret.yaml`)
4. try to do oauth flow again, should pick oauth config from the secret
